$OpenBSD$

- pthread_np.h doesn't seem to be required on OpenBSD
- OpenBSD supports pthread_self to get the current thread ID

Index: dlls/ntdll/server.c
--- dlls/ntdll/server.c.orig
+++ dlls/ntdll/server.c
@@ -31,7 +31,7 @@
 #ifdef HAVE_LWP_H
 #include <lwp.h>
 #endif
-#ifdef HAVE_PTHREAD_NP_H
+#if defined(HAVE_PTHREAD_NP_H) && !defined(__OpenBSD__)
 # include <pthread_np.h>
 #endif
 #include <signal.h>
@@ -1351,6 +1351,8 @@ static int get_unix_tid(void)
     mach_port_deallocate(mach_task_self(), ret);
 #elif defined(__NetBSD__)
     ret = _lwp_self();
+#elif defined(__OpenBSD__)
+    ret = syscall(SYS_getthrid);
 #elif defined(__FreeBSD__)
     long lwpid;
     thr_self( &lwpid );
